cabal-install-solver/src/Distribution/Solver/Types/PkgConfigDb.hs

```diff
@@ -63,10 +71,38 @@ readPkgConfigDb verbosity progdb = handle ioErrorHandler $ do
     case mpkgConfig of
       Nothing             -> noPkgConfig "Cannot find pkg-config program"
       Just (pkgConfig, _) -> do
-        pkgList <- lines <$> getProgramOutput verbosity pkgConfig ["--list-all"]
-        -- The output of @pkg-config --list-all@ also includes a description
-        -- for each package, which we do not need.
-        let pkgNames = map (takeWhile (not . isSpace)) pkgList
+        -- To prevent malformed Unicode in the descriptions from crashing cabal,
+        -- read without interpreting any encoding first. (#9608)
+        pkgList <- LBS.split (fromIntegral (ord '\n')) <$>
```

Comment:
This implementation doesn't seem very nice to me, it at least should be abstracted into it's own function.

* How about using `getProgramInvocationsIODataAndErrors`? Which doesn't crash when the `exitCode` is not `ExitSuccess`
* Could you use a parser abstraction to parse the input (you can use `Parsec` interface in `cabal-install-solver`)

In any case I don't think all this low level logic should be inlined here, so how about a new top level function `getPkgConfigPkgList` to encapsulate it all.

Reply:
I think using parsec to split a package list on newlines is extreme overkill. Additionally, I don't think this is a lot of logic to inline, just well documented logic. And finally its actually the _bulk_ of the function -- the existing top level function is _already_ the encapsulation.